src/core/Mixer.cpp

```diff
@@ -136,15 +136,12 @@ Mixer::Mixer( bool renderOnly ) :
 	// now that framesPerPeriod is fixed initialize global BufferManager
 	BufferManager::init( m_framesPerPeriod );
 
-	for( int i = 0; i < 3; i++ )
-	{
-		m_readBuf = (surroundSampleFrame*)
-			MemoryHelper::alignedMalloc( m_framesPerPeriod *
-						sizeof( surroundSampleFrame ) );
+	int outputBufferSize = m_framesPerPeriod * sizeof(surroundSampleFrame);
+	m_outputBufferRead = (surroundSampleFrame*)MemoryHelper::alignedMalloc(outputBufferSize);
+	m_outputBufferWrite = (surroundSampleFrame*)MemoryHelper::alignedMalloc(outputBufferSize);
```

Comment:
@DomClark @PhysSong do we want this as a `dynamic_cast`?

Reply:
It should be `static_cast` (example 10 on [this cppreference page](https://en.cppreference.com/w/cpp/language/static_cast)). `dynamic_cast` is only for casting inside inheritance hierarchy, and `reinterpret_cast` only for very special cases, like casting a `struct` int `int` if its first member is an `int`.

Reply:
Gotcha! I never remember which to use for casting from malloc calls instead of using new and such.